Gordon360/Services/HousingService.cs

```diff
@@ -537,5 +560,215 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
             _context.SaveChanges();
             return true;
         }
+
+        /// <summary>
+        /// update roommate imformation
+        /// </summary>
+        /// <param name="username"> The username for the user who complete the aplication form </param>
+        /// <param name="applicantion_id"> The ID of this application </param>
+        /// <param name="emailList"> A list of applicants' emails </param>
+        public async Task UpdateRoommateAsync(string username, string applicantion_id, string[] emailList)
+        {
+            var GordonID = _context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.gordon_id;
+            if (GordonID == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The applicant could not be found" };
+            }
+            var existApplicant = _context.Applicant.Where(x => x.ApplicationID == applicantion_id);
+            var existMaxYear = _context.Year.Where(x => x.ApplicationID == applicantion_id);
+            foreach (Applicant a in existApplicant)
+            {
+                _context.Applicant.Remove(a);
+            }
+            foreach (Year y in existMaxYear)
+            {
+                _context.Year.Remove(y);
+            }
+            List<string> yearList = new List<string>();
+            foreach (string e in emailList)
+            {
+                if (e != "")
+                {
+                    var newApplicant = new Applicant
+                    {
+                        ApplicationID = applicantion_id,
+                        Applicant1 = e
+                    }; ;
+                    await _context.Applicant.AddAsync(newApplicant);
+
+                    var student = _context.Student.FirstOrDefault(x => x.Email == e);
+                    if (student != null)
+                    {
+                        yearList.Add(student.Class);
+                    }
+                }
+            }
+            var newMaxYear = new Year
+            {
+                ApplicationID = applicantion_id,
+                Year1 = yearList.Max()
+            }; ;
```

Comment:
If you are going to keep this logic, why not just have one field on the application for `MaxStudentClassYear` or something which will be updated on application update/roommate change? It seems like there's a lot of overhead without need by inserting a row into a table every time roommates are updated.